### PR TITLE
KAFKA-17009: Add unit test to query nonexistent replica by describeReplicaLogDirs

### PR DESCRIPTION
**Pull request by m1a2st**
https://issues.apache.org/jira/browse/KAFKA-17009

add new test `testDescribeLogDirsWithNonExistReplica()`

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java [hunk 1]
```diff
@@ -6946,6 +6946,25 @@ public void testDescribeLogDirsPartialFailure() throws Exception {
...
+            env.kafkaClient().setNodeApiVersions(NodeApiVersions.create());
+            env.kafkaClient().prepareResponseFrom(
+                    new DescribeLogDirsResponse(
+                            new DescribeLogDirsResponseData().setResults(singletonList(
+                                    prepareDescribeLogDirsResult(tpr1, "/var/data/kafka0", 123456, 1, false)))),
+                    env.cluster().nodeById(brokerId));
+
+            DescribeReplicaLogDirsResult result = env.adminClient().describeReplicaLogDirs(asList(tpr1, tpr2));
+            Map<TopicPartitionReplica, KafkaFuture<DescribeReplicaLogDirsResult.ReplicaLogDirInfo>> values = result.values();
+            assertNull(values.get(tpr2).get().getCurrentReplicaLogDir());
```
Comment (chia7712) [inline clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java:6973]: could you verify the result of `tpr1`?

**Review by chia7712 — COMMENTED** [review]
@m1a2st thanks for this patch

File: clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java (hunk 1 above)
Comment (m1a2st) [reply clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java:6973]: Yes, I will add assert tp1 currentReplicaLogDir

File: clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java [hunk 2]
```diff
@@ -6946,6 +6946,28 @@ public void testDescribeLogDirsPartialFailure() throws Exception {
...
+            env.kafkaClient().prepareResponseFrom(
+                    new DescribeLogDirsResponse(
+                            new DescribeLogDirsResponseData().setResults(singletonList(
+                                    prepareDescribeLogDirsResult(tpr1, logDir, 123456, 1, false)))),
+                    env.cluster().nodeById(brokerId));
+
+            DescribeReplicaLogDirsResult result = env.adminClient().describeReplicaLogDirs(asList(tpr1, tpr2));
+            Map<TopicPartitionReplica, KafkaFuture<DescribeReplicaLogDirsResult.ReplicaLogDirInfo>> values = result.values();
+            
+            assertEquals(logDir, values.get(tpr1).get().getCurrentReplicaLogDir());
```
Comment (chia7712) [inline clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java:6969]: Could you please verify other getters also?

**Review by chia7712 — COMMENTED** [review]
@m1a2st thanks for completing test. One more thing: could you please update the docs of its public getters (`getCurrentReplicaLogDir`, `getCurrentReplicaOffsetLag`, `getFutureReplicaLogDir`, and `getFutureReplicaOffsetLag`). We can move the description of inner variable [0] to public scope

[0] https://github.com/apache/kafka/blob/7fe3374f7a6d956777b6b007300fa3fea6cf7118/clients/src/main/java/org/apache/kafka/clients/admin/DescribeReplicaLogDirsResult.java#L70

**Comment by m1a2st** [comment]
@chia7712, Thanks for your comments, PTAL

**Review by chia7712 — APPROVED** [review]
LGTM

**Pull request merged**